### PR TITLE
Improve on last used function

### PR DESCRIPTION
**Pull request by TonyLianLong**
This continues the PR #159 because Abizer has made useful suggestions to `last_used`.

### CODE REVIEW

File: ocflib/lab/stats.py [hunk 1]
```diff
@@ -140,6 +141,17 @@ def list_desktops(public_only=False):
         return [entry['attributes']['cn'][0] for entry in c.response]
 
 
+def last_used(hostname, password):
+    """Show the last used statistics for a computer."""
+    with get_connection(user='ocfstats-ro', password=password) as c:  # pylint: disable=E1124
```
Comment (abizer) [inline ocflib/lab/stats.py]: the signature for this function should be

```python
def last_used(hostname, ctx):
  query = "..."
  ctx.execute(...)
  return Session.from_row(c.fetchone())
```

i.e. build the MySQL connection object in the caller, not in the callee.

File: ocflib/lab/stats.py [hunk 2]
```diff
@@ -140,6 +141,17 @@ def list_desktops(public_only=False):
...
+
```
Comment (abizer) [inline ocflib/lab/stats.py]: nit: unnecessary whitespace

File: ocflib/lab/stats.py [hunk 3]
```diff
@@ -172,12 +184,8 @@ def from_hostname(cls, hostname, start, end):
 
     @classmethod
     def from_hostnames(cls, hostnames, start, end):
-        def add_ocf(hostname):
-            if not hostname.endswith('.ocf.berkeley.edu'):
-                return hostname + '.ocf.berkeley.edu'
-            return hostname
 
-        hostnames = tuple(map(add_ocf, hostnames))
+        hostnames = tuple(map(domain_from_hostname, hostnames))
```
Comment (abizer) [inline ocflib/lab/stats.py]: why do we call `domain_from_hostname` redundantly here and above? IMO any data we are passing into deeper layers of code should get sanitized at higher layers

File: ocflib/lab/stats.py (hunk 1 above)
Comment (TonyLianLong) [reply ocflib/lab/stats.py]: Sure, and in this case we don't have to consider the bug in pylint.

Comment (TonyLianLong) [reply ocflib/lab/stats.py]: But then it would be different from other functions in this file, which calls `get_connection`, and does it matter? Since other functions do not need external context to authenticate, I guess it's fine?

Comment (abizer) [reply ocflib/lab/stats.py]: it's fine, though in some cases we may want to consider updating their signatures as well. Depends on where they're used.
